### PR TITLE
Fix wrong calculation of stderr

### LINKED ISSUE #162: Error bar calculations, double check
Comments look strange.

in detector.py line 112
```
        if error_estimate != ErrorEstimate.none:
            self.error = np.nanstd(l, axis=0, ddof=1)  # stddev = sqrt( var / (n-1) )
```
comment a bit confusing. stddev = sqrt(1/(n-1)sum(x-x_mean)² = s where 

- s² is the "unbiased sample variance"
- s is the "corrected sample standard deviation"
- S = s/sqrt(N) is the "corrected standard deviation of the mean" (i.e. "standard error")
and then in line 317:

```
    #   stderr = stddev / sqrt(n)
    if len(input_file_list) > 1 and options.error == ErrorEstimate.stderr:
        first.error /= np.float64(first.counter)
```
```sqrt()``` missing?

### PR DESCRIPTION
**Pull request by nbassler**
Closes #162

### CODE REVIEW

File: pymchelper/detector.py [hunk 1]
```diff
@@ -129,7 +131,7 @@ def average_with_other(self, other_detector, error_estimate=ErrorEstimate.stderr
         self.data += delta / self.counter                      # mean += delta / n
         if error_estimate != ErrorEstimate.none:
             self._M2 += delta * (other_detector.data - self.data)  # M2 *= delta * (x - mean)
```
Comment (grzanka) [inline pymchelper/detector.py]: There is a typo in a comment to the right. It should be:
`M2 += delta * (x - mean)`
not
`M2 *= delta * (x - mean)`

File: pymchelper/detector.py [hunk 2]
```diff
@@ -129,7 +131,7 @@ def average_with_other(self, other_detector, error_estimate=ErrorEstimate.stderr
...
-            self.error = np.sqrt(self._M2 / (self.counter - 1))    # stddev = sqrt( var / (n-1) )
+            self.error = np.sqrt(self._M2 / (self.counter - 1))    # stddev = sqrt(1/(n-1)sum(x-<x>)**2)
```
Comment (grzanka) [inline pymchelper/detector.py]: I would add a comment line above explicitly mentioning that:
`self._M2 / (self.counter - 1)` holds a variance calculated using an online algorithm described above.
by taking square root of variance we obtain standard deviation which is then stored in `self.error` variable.

Comment (nbassler) [inline pymchelper/detector.py]: if you want to mention this, then let us go *all* the way and be most precise.

It is not the usual variance (var = sigma**2), but the *unbiased sample variance* ```s**2 = 1/(n-1)sum(x-<x>)**2```

Comment (nbassler) [inline pymchelper/detector.py]: can't you edit the comment as it pleases you, then I will comment on your comment to my comment.

(and please keep in mind where we waster our time... ;-)

Comment (nbassler) [inline pymchelper/detector.py]: btw, you already provide the link in the header a bit further upstream the code.

Comment (nbassler) [inline pymchelper/detector.py]: i have nothing more to add to the code, so I am done here.

Comment (grzanka) [inline pymchelper/detector.py]: I've extedned comments a bit

**Pull request merged**